CORE-9318 Separate DB users for automatic bootstrap and runtime

jacob-scott3141 opened the pull request.

blsemo committed:
NOTICK Extend test app and smoke tests for UTXO ledger (#3020)

* Add peekTransaction flow that uses the getInputStateAndRef and getOutputStateAndRef methods.
* Add a flow to evolve a demo state
* Smoke test for ledger evolution
* Add test for loading states from ledger tx

jacob-scott3141 committed:
seperate user and password for bootstrapping

davidcurrie commented on charts/corda/templates/_helpers.tpl:
```diff
@@ -561,6 +561,60 @@ Cluster DB credentials environment variables
+      {{- if .Values.bootstrap.db.cluster.username.valueFrom.secretKeyRef.name }}
+      name: {{ .Values.bootstrap.db.cluster.username.valueFrom.secretKeyRef.name | quote }}
+      key: {{ required "Must specify bootstrap.db.cluster.username.valueFrom.secretKeyRef.key" .Values.bootstrap.db.cluster.username.valueFrom.secretKeyRef.key | quote }}
+      {{- else }}
+      {{- if .Values.bootstrap.db.cluster.username.value }}
```
nit: I tend to format these as `{{- else if ... }}`

davidcurrie commented on charts/corda/templates/bootstrap-cluster-db-secret.yaml:
```diff
@@ -0,0 +1,19 @@
+{{- if or ( .Values.bootstrap.db.cluster.password.value) (.Values.bootstrap.db.cluster.username.value) }}
```
nit: you don't really need the parentheses here given they're only single values - you do have them then I tend to be more generous with the whitespace e.g. ` ( .Values.foo ) `
```suggestion
{{- if or .Values.bootstrap.db.cluster.password.value .Values.bootstrap.db.cluster.username.value }}

```

davidcurrie commented on charts/corda/values.yaml:
```diff
@@ -189,6 +189,31 @@ bootstrap:
           key: ""
   # Configuration for database bootstrap
   db:
+    cluster:
```
missing a doc comment for `bootstrap.db.cluster`

davidcurrie reviewed (commented):
A few nits but otherwise LGTM

jacob-scott3141 committed:
formatting changes

blsemo committed:
Fix detekt

jacob-scott3141 committed:
add bootstrap user to e2e tests

blsemo committed:
Fix misplaced service field

yarivus reviewed (dismissed):
LGTM

simon-johnson-r3 committed:
CORE-9716 Extend database plugin to support Liquibase tracking tables (#3057)

- Schemas are no longer hardcoded in the Corda 5 codebase, either in SQL queries or JPA Entity class declarations.
- The CONFIG schema (which can now be named anything a customer likes) is encoded into the JDBC URL passed to db aware workers on boot up. All other schemas: RBAC, CRYPTO, and the dynamic ones added for each virtual node, are encoded into JDBC URLs in the config's db_connection table (there is also a new row in this table, a JDBC URL which tells the Db Worker where to put Vnode dynamic schemas) which is populated as part of bootstrapping. Thus workers are now entirely unaware of schema names, they will simply understand that tables for some data store are at the end of some JDBC URL.
- Adds support for Liquibase version tracking tables in the Corda cluster database(s) to the database plugin. The plugin can be used to generate not just bootstrap SQL in this mode, but also SQL required to upgrade the database to any current version via Liquibase. New parameters are added to provide a JDBC URL, username and password in the same manner other plugins used by bootstrapping do in order to keep the InitContainers' configuration aligned with each other. The tool reverts to its previous offline behaviour if the JDBC URL is omitted.
- The Corda Helm chart is modified to support online mode for bootstrapping.

jacob-scott3141 committed:
granted permissions to DB_CLUSTER_USERNAME

yarivus reviewed (dismissed):
LGTM

yarivus reviewed (dismissed):
LGTM

jacob-scott3141 committed:
fixed bash scripts and merged changes with recent PRs
